taskcluster/app_services_taskgraph/build_config.py

```diff
@@ -44,7 +42,7 @@ def get_version(params):
     else:
         return version
 
-@memoize
+@functools.lru_cache(maxsize=None)
```

Comment:
You could just make this `@functools.cache`, it's an alias for `lru_cache(maxsize=None)`. It was introduced in 3.9, but the Decision image should be using 3.11 these days.